### PR TITLE
add general caching via callback fn

### PR DESCRIPTION
**Pull request by pgorecki**

### CODE REVIEW

File: imports/api/collectionsCount/server/hmisCounts.js [hunk 1]
```diff
@@ -5,6 +5,15 @@ class HmisCounts {
     this.userId = userId;
   }
 
+  getClientsCount() {
+    const hc = HmisClient.create(this.userId);
+    const schemas = ['v2017', 'v2016', 'v2015', 'v2014'];
+    return schemas.reduce(
+      (count, schema) => count + hc.api('client').getClients(schema).length,
```
Comment (art4ride) [inline imports/api/collectionsCount/server/hmisCounts.js]: Maybe we should send clients with limit=1 and look for pagination.total data

File: imports/startup/client/routes/clients.js [hunk 2]
```diff
@@ -162,6 +162,7 @@ Router.route(
         } else {
           if (hmisClient) {
             Bert.alert('This client has already been surveyed', 'danger', 'growl-top-right');
+            // XXXX
```
Comment (art4ride) [inline imports/startup/client/routes/clients.js]: What does it stand for?

File: imports/api/collectionsCount/server/publications.js [hunk 3]
```diff
@@ -21,21 +21,22 @@ function publishCounts(collection, collectionName, publication) {
...
-Meteor.publish('collectionsCount', function publishCollectionCount() {
+Meteor.publish('collectionsCount', function publishCollectionCount(forceReload = true) {
   if (!this.userId) {
     return [];
   }
 
-  const handles = [];
-  const self = this;
-
-  handles.push(publishCounts(PendingClients, 'clients', this));
```
[the remark this answers is not in the document]
Comment (pgorecki) [reply imports/api/collectionsCount/server/publications.js]: we are not using pending clients right now, you cannot do anything useful with them either

File: imports/api/collectionsCount/server/hmisCounts.js (hunk 1 above)
Comment (pgorecki) [reply imports/api/collectionsCount/server/hmisCounts.js]: +1

File: imports/startup/client/routes/clients.js (hunk 2 above)
Comment (pgorecki) [reply imports/startup/client/routes/clients.js]: to be removed

**Pull request merged**